File: RentACarApplication/src/app/car/dtos/search-car.dto.ts
import { CategoryEnum } from "../../enums/category-enum";
import { EngineEnum } from "../../enums/engine-enum";
import { RegionEnum } from "../../enums/region-enum";
import { TransmissionEnum } from "../../enums/transmission-enum";
import { CarImage } from "./image.dto";

export class SearchCarDto{
  id: number
  make: string;
  model: string;  
  price: number;
  year:number;
  region: RegionEnum; 
  transmission: TransmissionEnum;
  engine: EngineEnum;
  category: CategoryEnum
  isRented: boolean;
  imageFiles: File;
  images: CarImage[];
}

File: RentACarApp/Dtos/CarDtos/SearchCarDto.cs
using RentACarApp.Database.Enums;
using RentACarApp.Database.Models;

namespace RentACarApp.Dtos.CarDtos
{
    public class SearchCarDto
    {
        public string Make { get; set; }

        public string Model { get; set; }

        public int Year { get; set; }

        public decimal Price { get; set; }

        public EngineEnum? Engine { get; set; }

        public TransmissionEnum? Transmission { get; set; }

        public RegionEnum? Region { get; set; }

        public CategoryEnum? Category { get; set; }

        public bool IsRented { get; set; }

        public List<IFormFile> ImageFiles { get; set; }

        public List<Image> Images { get; set; } = new List<Image>();

    }
}
